validation/open_boundaries/cylinder.jl

```diff
@@ -1,127 +1,214 @@
+"""
+function drag(model;
+              bounding_box = (-1, 3, -2, 2),
+              ν = 1e-3)
 
-# add extra downstream distance to see if the solution near the cylinder changes
-extra_downstream = 0
+    u, v, _ = model.velocities
 
-cylinder = Cylinder(; D)
+    uᶜ = Field(@at (Center, Center, Center) u)
+    vᶜ = Field(@at (Center, Center, Center) v)
 
-x = (-5, 5 + extra_downstream) .* D
-y = (-5, 5) .* D
+    xc, yc, _ = nodes(uᶜ)
 
-Ny = Int(10 / resolution)
-Nx = Ny + Int(extra_downstream / resolution)
+    i₁ = findfirst(xc .> bounding_box[1])
+    i₂ = findlast(xc .< bounding_box[2])
 
-ν = U * D / Re
+    j₁ = findfirst(yc .> bounding_box[3])
+    j₂ = findlast(yc .< bounding_box[4])
 
-closure = ScalarDiffusivity(;ν, κ = ν)
+    uₗ² = Field(uᶜ^2, indices = (i₁, j₁:j₂, 1))
+    uᵣ² = Field(uᶜ^2, indices = (i₂, j₁:j₂, 1))
 
-grid = RectilinearGrid(architecture; topology = (Bounded, Periodic, Flat), size = (Nx, Ny), x, y)
+    uvₗ = Field(uᶜ*vᶜ, indices = (i₁:i₂, j₁, 1))
+    uvᵣ = Field(uᶜ*vᶜ, indices = (i₁:i₂, j₂, 1))
 
-@inline u(y, t, U) = U * (1 + 0.01 * randn())
+    ∂₁uₗ = Field(∂x(uᶜ), indices = (i₁, j₁:j₂, 1))
+    ∂₁uᵣ = Field(∂x(uᶜ), indices = (i₂, j₁:j₂, 1))
 
-u_boundaries = FieldBoundaryConditions(east = FlatExtrapolationOpenBoundaryCondition(),
-                                       west = OpenBoundaryCondition(u, parameters = U))
+    ∂ₜuᶜ = Field(@at (Center, Center, Center) model.timestepper.Gⁿ.u)
 
-v_boundaries = FieldBoundaryConditions(east = GradientBoundaryCondition(0),
-                                       west = GradientBoundaryCondition(0))
+    ∂ₜu = Field(∂ₜuᶜ, indices = (i₁:i₂, j₁:j₂, 1))
 
-Δt = .3 * resolution / U
+    p = model.pressures.pNHS
 
-u_forcing = Relaxation(; rate = 1 / (2 * Δt), mask = cylinder)
-v_forcing = Relaxation(; rate = 1 / (2 * Δt), mask = cylinder) 
+    ∫∂ₓp = Field(∂x(p), indices = (i₁:i₂, j₁:j₂, 1))
 
-model = NonhydrostaticModel(; grid, 
-                              closure, 
-                              forcing = (u = u_forcing, v = v_forcing),
-                              boundary_conditions = (u = u_boundaries, v = v_boundaries))
+    a_local = Field(Integral(∂ₜu))
 
-@info "Constructed model"
+    a_flux = Field(Integral(uᵣ²)) - Field(Integral(uₗ²)) + Field(Integral(uvᵣ)) - Field(Integral(uvₗ))
 
-# initial noise to induce turbulance faster
-set!(model, u = U, v = (x, y) -> randn() * U * 0.01)
+    a_viscous_stress = 2ν * (Field(Integral(∂₁uᵣ)) - Field(Integral(∂₁uₗ)))
 
-@info "Set initial conditions"
+    a_pressure = Field(Integral(∫∂ₓp))
 
-simulation = Simulation(model; Δt = Δt, stop_time = 300)
+    return a_local + a_flux + a_pressure - a_viscous_stress
+end
 
-wizard = TimeStepWizard(cfl = 0.3)
+Re = 1000
+#=
+if Re <= 100
+    Ny = 512 
+    Nx = Ny
+elseif Re <= 1000
+    Ny = 2^11
+    Nx = Ny
+elseif Re == 10^4
+    Ny = 2^12
+    Nx = Ny
+elseif Re == 10^5
+    Ny = 2^13
+    Nx = Ny
+elseif Re == 10^6
+    Ny = 3/2 * 2^13 |> Int
+    Nx = Ny
+end
+=#
```

Comment:
There are several commented portions of this script. Maybe you forgot to clean up?